PCbuild/find_python.bat

```diff
@@ -34,10 +34,10 @@
 @if exist "%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe" (set PYTHON="%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe") & (set _Py_Python_Source=found in externals directory) & goto :found
 
 @rem If HOST_PYTHON is recent enough, use that
-@if NOT "%HOST_PYTHON%"=="" @%HOST_PYTHON% -Ec "import sys; assert sys.version_info[:2] >= (3, 8)" >nul 2>nul && (set PYTHON="%HOST_PYTHON%") && (set _Py_Python_Source=found as HOST_PYTHON) && goto :found
+@if NOT "%HOST_PYTHON%"=="" @%HOST_PYTHON% -Ec "import sys; assert sys.version_info[:2] >= (3, 10)" >nul 2>nul && (set PYTHON="%HOST_PYTHON%") && (set _Py_Python_Source=found as HOST_PYTHON) && goto :found
```

Comment:
This narrows the bootstrap Python windows quite a bit. If people want to use older Pythons, somebody has to rewrite the generator without match/case.

Reply:
I've already been tempted to write a `match()` function that returns an object supporting `m.case("match pattern")` ;) but not tempted enough to do it here.

However, we'll need to add an earlier check or all existing developers will fail to run the script without a decent error. I'll post it as a suggestion